src/main/java/io/cdap/plugin/spark/dynamic/ScalaSparkSink.java

```diff
@@ -64,7 +67,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
 
   @Override
   public void prepareRun(SparkPluginContext sparkPluginContext) throws Exception {
-    // no-op
+    Schema schema = sparkPluginContext.getInputSchema();
+    if (schema != null && schema.getFields() != null) {
+      recordLineage(sparkPluginContext, config.scalaCode, schema, // TODO:better reference name here?
```

Comment:
this shouldn't be put in cdap, also if reference name is not provided, should add that to the config instead of using this scalaCode. this will use the entire code block as the dataset name

Reply:
I just added it to config in this class. Can you review again?